comp/dogstatsd/server/enrich.go

```diff
@@ -81,6 +81,8 @@ func extractTagsMetadata(tags []string, originFromUDS string, originFromMsg []by
 			cardinality = tag[len(CardinalityTagPrefix):]
 		} else if strings.HasPrefix(tag, jmxTagPrefix) {
 			metricSource = metrics.MetricSourceJmxCustom
+			tags[n] = tag
```

Comment:
This looks good, but maybe it'd be clearer to have this "save tag" action only specified once? 
What do you think of this approach using `continue` and no `else` block?
I'm torn because it also seems a bit more error-prone (have to explicitly add a `continue` to each block where we _don't_ want the tag added)


```
	for _, tag := range tags {
		if strings.HasPrefix(tag, hostTagPrefix) {
			host = tag[len(hostTagPrefix):]
            continue
		} else if strings.HasPrefix(tag, entityIDTagPrefix) {
			originFromTag = tag[len(entityIDTagPrefix):]
             continue
		} else if strings.HasPrefix(tag, CardinalityTagPrefix) {
			cardinality = tag[len(CardinalityTagPrefix):]
            continue
		} else if strings.HasPrefix(tag, jmxTagPrefix) {
			metricSource = metrics.MetricSourceJmxCustom
             continue
		}
	    tags[n] = tag
		n++
		
	}
```


Up to you, just a suggestion.